Library/Homebrew/cmd/update-report.rb

```diff
@@ -209,8 +209,12 @@ def report
         end
         @report[:M] << tap.formula_file_to_name(src)
       when /^R\d{0,3}/
-        @report[:D] << tap.formula_file_to_name(src) if tap.formula_file?(src)
-        @report[:A] << tap.formula_file_to_name(dst) if tap.formula_file?(dst)
+        src_full_name = tap.formula_file_to_name(src)
+        dst_full_name = tap.formula_file_to_name(dst)
+        # Don't report formulae that are moved within a tap but not renamed
+        next if src_full_name == dst_full_name
+        @report[:D] << src_full_name if tap.formula_file?(src)
+        @report[:A] << dst_full_name if tap.formula_file?(dst)
```

Comment:
There's a very subtle problem with these two lines. If formulae happen to be moved within a tap *and* a formula gets renamed before or afterwards (thus `src_full_name != dst_full_name`) and the update covers both operations, then the report will look something like this (I experimented a bit with my local copy of the `homebrew/gui` tap):

```
Updated 1 tap (homebrew/gui).
==> New Formulae
homebrew/gui/xchat2
```

But it should have looked something like this:

```
Updated 1 tap (homebrew/gui).
==> Renamed Formulae
homebrew/gui/xchat -> homebrew/gui/xchat2
```

The reason is that `if tap.formula_file?(src)` will always be `false` as it checks against the current `Tap#formula_dir`. It's admittedly a rare corner case, but this should solve it:

```ruby
if tap.formula_file?(dst)
  @report[:D] << src_full_name
  @report[:A] << dst_full_name
end
```